### PR TITLE
switch to Fetch API

### PR DESCRIPTION
**Pull request by mccalluc**
Towards #287.
- I probably should have branched from develop, right?
- How do I test things on this project?
- The two methods are now almost identical, except for `json` and `text` on one line, and the mime type: Introduce a wrapper method?

### CODE REVIEW

File: app/scripts/services/tile-proxy.js [hunk 1]
```diff
@@ -532,36 +527,54 @@ export const tileDataToPixData = (
...
+      if (response.ok) {
+        response.text().then(t => {
+          callback(null, t);
+          requestsInFlight -= 1;
+        })
+      } else {
+        // Do something?
+      }
+    })
+    .catch(error => {
```
Comment (flekschas) [inline app/scripts/services/tile-proxy.js]: Flight counter is not updated. I'd suggest using `finally` to update it.

Comment (flekschas) [reply app/scripts/services/tile-proxy.js]: I am done a super similar update here https://github.com/hms-dbmi/higlass/pull/307
Maybe we can merge the two efforts. :)

**Comment by flekschas** [comment]
This is very likely going to clash with https://github.com/hms-dbmi/higlass/pull/307 where I replace d3text with `fetch`. Maybe we can merge the two efforts :)

File: app/scripts/services/tile-proxy.js (hunk 1 above)
Comment (mccalluc) [reply app/scripts/services/tile-proxy.js]: How about if I make a branch off of your branch, and just duplicate what you've done for `text` for `json`? Thanks!

**Pull request closed without merging**